spec/classes/mod/authnz_ldap_spec.rb

```diff
@@ -41,50 +41,53 @@
   end # Debian
 
   context 'default configuration with parameters on a RedHat OS' do
-    let :facts do
-      {
-        osfamily: 'RedHat',
-        operatingsystemrelease: '6',
-        id: 'root',
-        kernel: 'Linux',
-        operatingsystem: 'RedHat',
-        path: '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
-        is_pe: false,
-      }
-    end
-
-    it { is_expected.to contain_class('apache::params') }
-    it { is_expected.to contain_class('apache::mod::ldap') }
-    it { is_expected.to contain_apache__mod('authnz_ldap') }
-
-    context 'default verify_server_cert' do
-      it { is_expected.to contain_file('authnz_ldap.conf').with_content(%r{^LDAPVerifyServerCert On$}) }
-    end
-
-    context 'verify_server_cert = false' do
-      let(:params) { { verify_server_cert: false } }
-
-      it { is_expected.to contain_file('authnz_ldap.conf').with_content(%r{^LDAPVerifyServerCert Off$}) }
-    end
-
-    context 'verify_server_cert = wrong' do
-      let(:params) { { verify_server_cert: 'wrong' } }
-
-      it 'raises an error' do
-        expect { is_expected.to raise_error Puppet::Error }
+    on_supported_os.each do |os, facts|
```

Comment:
Recently the recommended pattern changed to use `os_facts` so you don't have an overlapping variable with a let block.
```suggestion
    on_supported_os.each do |os, os_facts|
```